### PR TITLE
Add an optional repeat param to UpdateJob

### PR DESCRIPTION
**Pull request by tgolen**
# Test

```
CreateJob
name: repeattest1
data: {1:1}
repeat: finished, +1 minute

200 OK
commitCount: 356
nodeName: bedrock
processingTime: 7
totalTime: 9
waitTime: 1
Content-Length: 11

{"jobID":9}

QueryJob
jobID: 9

200 OK
commitCount: 356
nodeName: bedrock
processingTime: 3
totalTime: 4
waitTime: 0
Content-Length: 172

{"created":"2017-01-26 18:43:00","data":"{1:1}","jobID":9,"lastRun":"","name":"repeattest1","nextRun":"2017-01-26 18:43:00","repeat":"FINISHED, +1 MINUTE","state":"QUEUED"}

#  Test against regressions by not using the repeat param
UpdateJob
jobID: 9
data: {2:2}

200 OK
commitCount: 357
nodeName: bedrock
processingTime: 3
totalTime: 4
waitTime: 0
Content-Length: 0

QueryJob
jobID: 9

200 OK
commitCount: 357
nodeName: bedrock
processingTime: 5
totalTime: 5
waitTime: 0
Content-Length: 172

{"created":"2017-01-26 18:43:00","data":"{2:2}","jobID":9,"lastRun":"","name":"repeattest1","nextRun":"2017-01-26 18:43:00","repeat":"FINISHED, +1 MINUTE","state":"QUEUED"}

# Test the new param that the repeat column is updated properly
UpdateJob
jobID: 9
data: {3:3}
repeat: started, +2 minute

200 OK
commitCount: 358
nodeName: bedrock
processingTime: 7
totalTime: 9
waitTime: 2
Content-Length: 0

QueryJob
jobID: 9

200 OK
commitCount: 358
nodeName: bedrock
processingTime: 3
totalTime: 3
waitTime: 0
Content-Length: 171

{"created":"2017-01-26 18:43:00","data":"{3:3}","jobID":9,"lastRun":"","name":"repeattest1","nextRun":"2017-01-26 18:43:00","repeat":"STARTED, +2 MINUTE","state":"QUEUED"}


```

Signed-off-by: Tim Golen <tgolen@expensify.com>

### CODE REVIEW

**Comment by tgolen** [comment]
Woo! A new era has begun. I'm officially a c++ developer now :grin:

File: plugins/Jobs.cpp [hunk 1]
```diff
@@ -367,8 +372,9 @@ bool BedrockPlugin_Jobs::processCommand(BedrockNode* node, SQLite& db, BedrockNo
         // Update the data
         if (!db.write("UPDATE jobs "
                       "SET data=" +
-                      SQ(request["data"]) + " "
-                                            "WHERE jobID=" +
+                      SQ(request["data"]) + " " +
+                      (request.isSet("repeat") ? ", repeat=" + SQ(SToUpper(request["repeat"])) : "") +
```
Comment (iwiznia) [inline plugins/Jobs.cpp]: I think this needs to be ` (request.isSet("repeat") ? ", repeat=" + SQ(SToUpper(request["repeat"])) : "''")` If you don't pass repeat, wouldn't you end up with: ` repeat = WHERE jobID = 1`, right?

**Comment by quinthar** [comment]
Isn't that good?  I'm not following your concern, can you restate it?

File: plugins/Jobs.cpp (hunk 1 above)
Comment (tgolen) [inline plugins/Jobs.cpp]: No. How could you get `, repeat=` without matching the truth test of the ternary? (also, this is identical to other code in this file that uses a ternary to update optional information)

Comment (iwiznia) [inline plugins/Jobs.cpp]: Oh, shit, sorry, misread the parenthesis. Forget me, I must probably be still drunk.

**Pull request merged**